### ISSUE TITLE
MPS simulator gives incorrect values for post-measure instructions

### ISSUE THREAD

**Issue by chriseclectic** [Qiskit/qiskit-aer#370]
### Informations

- **Qiskit Aer version**:
- **Python version**:
- **Operating system**:

### What is the current behavior?

If an instruction is placed after final measurement the MPS simulator sometimes gives incorrect results. This final result shouldn't change the outcome of measurements that have already happened.

### Steps to reproduce the problem

The following circuit:
```python
circuit = QuantumCircuit(2, 2)
circuit.x([0, 1])
circuit.measure([0, 1], [0, 1])
circuit.barrier([0, 1])
circuit.x([1])

job = execute(circuit, Aer.get_backend("qasm_simulator"),
              backend_options={"method": "matrix_product_state"})
result = job.result()
result.get_counts(0)
```
Returns counts `{'10': 508, '11': 516}`. 

### What is the expected behavior?

It should return counts `{'11': 1024}`.

**Comment by merav-aharoni** [comment]
There was a mistake in the initialization of the MPS data structure when it is called during multiple shots - the data structure was not cleared. Fixed in PR 390.

**Comment by chriseclectic** [comment]
Great! I think fixing this will fix a lot of the tests that were failing.

**Comment by chriseclectic** [comment]
Fixed in #393
